### PR TITLE
EDU-597 Add parent display name

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-597

### CODE REVIEW

File: src/services/storage-service.js [hunk 1]
```diff
@@ -134,27 +134,65 @@ export default class StorageService {
...
+        portableUrl: `cdn://${encodedObjectSegments.join('/')}`,
         createdOn: isDirectory ? null : obj.lastModified,
         type: isDirectory ? CDN_OBJECT_TYPE.directory : CDN_OBJECT_TYPE.file,
         size: isDirectory ? null : obj.size
       };
     });
 
-    return mappedObjects;
+    return { parentDirectory, currentDirectory, objects };
   }
```
Comment (adrianaluchian) [inline src/services/storage-service.js:196]: This method is decomposing and recomposing paths alot, feels like a test would be helpful here 🤔

**Comment by adrianaluchian** [comment]
The upper level navigation actually looks nice in the list view, we'll just need a better icon so it's good for the grid view as well

**Comment by ahelmberger** [comment]
> The upper level navigation actually looks nice in the list view, we'll just need a better icon so it's good for the grid view as well.
> 
> How would this look like to you? Screen Shot 2022-06-08 at 13 50 44

Agree, that looks a lot nicer. Did you create this as an SVG or just manipulate the pixels (i.e. can you provide me yours or should a re-create it :smiling_imp:)?

File: src/services/storage-service.js (hunk 1 above)
Comment (ahelmberger) [reply src/services/storage-service.js:196]: True, it's the one function in this service that has no tests, I will add some.

**Comment by adrianaluchian** [comment]
> > The upper level navigation actually looks nice in the list view, we'll just need a better icon so it's good for the grid view as well.
> > How would this look like to you? Screen Shot 2022-06-08 at 13 50 44
> 
> Agree, that looks a lot nicer. Did you create this as an SVG or just manipulate the pixels (i.e. can you provide me yours or should a re-create it 😈)?

Sure, as you're ok with it I've committed it to the assets repo, along with a version for the files list view (haven't checked how that one looks but feel free to use, adjust or delete it).

**Comment by ahelmberger** [comment]
> > [...]
> > Agree, that looks a lot nicer. Did you create this as an SVG or just manipulate the pixels (i.e. can you provide me yours or should a re-create it smiling_imp)?
> 
> Sure, as you're ok with it I've committed it to the assets repo, along with a version for the files list view (haven't checked how that one looks but feel free to use, adjust or delete it).

Thanks so much :pray:

File: src/services/storage-service.js (hunk 1 above)
Comment (ahelmberger) [reply src/services/storage-service.js:191]: Added, thanks for the suggestion, I actually found some edge case problems.

**Comment by ahelmberger** [comment]
> > [...]
> > Agree, that looks a lot nicer. Did you create this as an SVG or just manipulate the pixels (i.e. can you provide me yours or should a re-create it smiling_imp)?
> 
> Sure, as you're ok with it I've committed it to the assets repo, along with a version for the files list view (haven't checked how that one looks but feel free to use, adjust or delete it).

Added and adjusted the new icons :+1:

**Pull request merged**